cmd/httpd/handlers/auth.go

```diff
@@ -0,0 +1,169 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/ibliuchak/auth/internal/platform/storage"
+
+	"github.com/ibliuchak/auth/internal/tokens"
+
+	"golang.org/x/crypto/bcrypt"
+
+	"github.com/ibliuchak/auth/internal/users"
+
+	"github.com/rs/zerolog"
+)
+
+type Auth struct {
+	logger zerolog.Logger
+	users  users.Users
+	tokens tokens.Tokens
+}
+
+func NewAuth(logger zerolog.Logger, users users.Users, tokens tokens.Tokens) *Auth {
+	return &Auth{logger: logger, users: users, tokens: tokens}
+}
+
+type loginRequest struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+type loginResponse struct {
+	Token string `json:"token"`
+}
+
+type validateTokenRequest struct {
+	Token string `json:"token"`
+}
+
+func (a *Auth) Login(w http.ResponseWriter, r *http.Request) {
+	var request loginRequest
+
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&request)
+	if err != nil {
+		a.logger.Error().Err(err).Msg("Unable to decode login data")
+
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	user, err := a.users.GetUserByEmail(request.Email)
+	if err != nil {
+		if err == storage.ErrUserNotFound {
+			a.logger.Error().Err(err).Str("email", request.Email).Msg("user not found")
+
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		a.logger.Error().Err(err).Str("email", request.Email).Msg("Can't get user")
+
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// compare request password and record password
+	err = bcrypt.CompareHashAndPassword([]byte(user.Password), []byte(request.Password))
+	if err != nil {
+		a.logger.Error().Err(err).Str("email", user.Email).Msg("Bad password")
+
+		w.WriteHeader(http.StatusForbidden)
+		return
+	}
+
+	a.logger.Info().Str("email", user.Email).Msg("Successful login")
+
+	expiration := time.Now().Add(30 * time.Minute)
```

Comment:
Please move 30*time.Minute to const and change to 1 hour

Reply:
ticket for this has been created #6